Tests/AlicerceTests/AutoLayout/ConstraintGroupToggleTestCase.swift

```diff
@@ -0,0 +1,254 @@
+import XCTest
+@testable import Alicerce
+
+
+class ConstraintGroupToggleTestCase: BaseConstrainableProxyTestCase {
+
+    private enum TestConstraintGroupKey: Hashable {
+        case first
+        case second
+        case third
+    }
+
+    func testNoInitialConstraint_WithEnumKey_ShouldNotActivateConstraints() {
+
+        var constraint1: NSLayoutConstraint!
+        var constraint2: NSLayoutConstraint!
+        var constraint3: NSLayoutConstraint!
+
+        let _ = ConstraintGroupToggle<TestConstraintGroupKey> { [weak self] in
+
+            guard let self = self else { return nil }
+
+            switch $0 {
+            case .first:
+                return constrain(self.host, self.view0) { host, view0 in
+                   constraint1 = view0.bottom(to: host, offset: -100)
+                }
+
+            case .second:
+                return constrain(self.host, self.view0) { host, view0 in
+                   constraint2 = view0.top(to: host, offset: -100)
+                }
+
+            case .third:
+                return constrain(self.host, self.view0) { host, view0 in
+                   constraint3 = view0.leading(to: host, offset: -100)
+                }
+            }
+        }
+
+        XCTAssertNil(constraint1)
+        XCTAssertNil(constraint2)
+        XCTAssertNil(constraint3)
+    }
+
+    func testInitialConstraint_WithEnumKey_ShouldActivateFirstConstraint() {
+
+        var constraint1: NSLayoutConstraint!
+        var constraint2: NSLayoutConstraint!
+        var constraint3: NSLayoutConstraint!
+
+        let _ = ConstraintGroupToggle(initial: TestConstraintGroupKey.first) { [weak self] in
+
+            guard let self = self else { return nil }
+
+            switch $0 {
+            case .first:
+                return constrain(self.host, self.view0) { host, view0 in
+                   constraint1 = view0.bottom(to: host, offset: -100)
+                }
+
+            case .second:
+                return constrain(self.host, self.view0) { host, view0 in
+                   constraint2 = view0.top(to: host, offset: -100)
+                }
+
+            case .third:
+                return constrain(self.host, self.view0) { host, view0 in
+                   constraint3 = view0.leading(to: host, offset: -100)
+                }
+            }
+        }
+
+        XCTAssert(constraint1.isActive)
+        XCTAssertNil(constraint2)
+        XCTAssertNil(constraint3)
+    }
+
+    func testActivateOneConstraint_WithEnumKey_ShouldDeactivateFirstConstraintAndActivateSecondConstraint() {
+
+        var constraint1: NSLayoutConstraint!
+        var constraint2: NSLayoutConstraint!
+        var constraint3: NSLayoutConstraint!
+
+        let group = ConstraintGroupToggle(initial: TestConstraintGroupKey.first) { [weak self] in
```

Comment:
Maybe name this `groupToggle` instead of just `group` as it's a toggle of constraint groups and not just one constraint group.

Also these constraints and toggle are the same in all or almost all the tests so maybe they can  be set up in the `XCTest.setUp` override.

Reply:
will do 👍